### PR TITLE
Streamline creation of window ENV object

### PR DESCRIPTION
**Pull request by gcampo88**
Streamline the creation of window ENV object to solve env var errors

### CODE REVIEW

**Comment by gcampo88** [comment]
@cavis & @kookster want to take a look and let me know your thoughts?

File: server/config.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      } else if (isNaN(val) || val == '') {
+        val = `'${val}'`;
+      } else {
+        val = val;
+      }
+      envVar[key] = val
+    }
+  }
+
+  // remove the old dotenv.js, just in case
```
Comment (cavis) [inline server/config.js]: I think this is unnecessary - take `assets/dotenv.js` out of the gitignore if you want, but no need to explicitly nuke it.

File: server/config.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  // safe parse .env
+  let env = {};
+  try {
+    let dottext = fs.readFileSync(`${__dirname}/../.env`);
+    env = dotenv.parse(dottext);
+  } catch (err) {}
+
+  // pull overrides out of process.env and .env
+  let envVar = {"sample": "works"};
```
Comment (cavis) [inline server/config.js]: Test code?

File: server/config.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  for (let key of keys) {
+    if (process.env[key] !== undefined || (env[key] && env[key] !== '')) {
+      let val = process.env[key] || env[key];
+      if (['true', 'false', 'null', 'undefined'].indexOf(val) > -1) {
+        val = val;
+      } else if (isNaN(val) || val == '') {
```
Comment (cavis) [inline server/config.js]: Close, but I think slightly off.  The previous code was attempting to manually json-encode these to write the ENV-string.  Here you're returning a plain javascript object with unencoded values.

For instance, if you had `LOGO_NAME=prx` in your `.env`, then for this loop you'd have `key = "LOGO_NAME";` and `val = "prx"`.  Which would result in quoting the value to `"'prx'"` on the next line.

I had this same problem in the current Publish codebase, and [solved it here](https://github.com/PRX/publish.prx.org/blob/master/lib/util.js#L11).  (Sorry for not remembering that earlier!)  A good rainy day project would be to encapsulate all these server/env-parsing scripts into a `prx-ng2-server` npm package, or something.  But we should probably wait till Play is on webpack to take that on.

File: server/dist.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ const PORT = 3000;
 // pre-render
 const index = pug.renderFile('./src/index.pug', {
   dist: true,
-  // config: config.windowEnv()
+  config: config.windowEnv()
```
Comment (cavis) [inline server/dist.js]: You also need to add this to the dev server script.  I believe that `res.render('./src/index')` in there is calling to the pug renderer under the hood.  So you just need to pass a 2nd argument with `{config: config.windowEnv()}` to it, like they do in the [express docs](https://expressjs.com/en/guide/using-template-engines.html).

File: package.json [hunk 5]
```diff
@@ -29,7 +29,7 @@
     "prestart:dist": "",
     "start": "if [ $NODE_ENV='development' ]; then npm run start:dev; else npm run start:dist; fi",
     "start:dev": "node bin/mkenv.js && node ./server/dev.js",
```
Comment (cavis) [inline package.json]: Can also change the dev server to do the same logic, and delete the `bin` directory entirely.

File: server/config.js (hunk 2 above)
Comment (gcampo88) [inline server/config.js]: well I feel 🐑 ish!

File: server/config.js (hunk 3 above)
Comment (gcampo88) [inline server/config.js]: 👍

File: server/config.js (hunk 1 above)
Comment (gcampo88) [inline server/config.js]: 👍

**Comment by gcampo88** [comment]
round 2! @cavis I fixed the env parsing and made the dev server startup mimic the production server startup. LMK what you think!

**Pull request merged**